test/sai_test/sai_tunnel_test.py

```diff
@@ -88,7 +88,130 @@ def setUp(self):
     def tearDown(self):
         T0TestBase.tearDown(self)
 
-class IPInIPTunnelDecapv4Inv4Test(IpInIpTnnnelBase):
+class BasicIPInIPTunnelEncapv4Inv4Test(IpInIpTnnnelBase):
```

Comment:
We can use base class like, IpInIpTnnnelBase, that's good.
For common method reference and reuse, Please add a tunnel configer to organize the tunnel configurations.
Please keep the code structure well, like abstract the method for different purpose and parameterize the method for reference for reuse.